data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndRawSpanTest.java

```diff
@@ -126,13 +148,24 @@ private void sendExportTraceServiceRequestToSource(final int port, final ExportT
 
     private List<Map<String, Object>> getSourcesFromSearchHits(final SearchHits searchHits) {
         final List<Map<String, Object>> sources = new ArrayList<>();
-        searchHits.forEach(hit -> sources.add(hit.getSourceAsMap()));
+        searchHits.forEach(hit -> {
+            Map<String, Object> source = hit.getSourceAsMap();
+            // Elasticsearch API identifies Number type by range, need to convert to Long
+            if (source.containsKey(TraceGroupWrapper.TRACE_GROUP_DURATION_IN_NANOS_FIELD)) {
+                final Long durationInNanos = ((Number) source.get(TraceGroupWrapper.TRACE_GROUP_DURATION_IN_NANOS_FIELD)).longValue();
+                source.put(TraceGroupWrapper.TRACE_GROUP_DURATION_IN_NANOS_FIELD, durationInNanos);
+            }
+            sources.add(source);
+        });
         return sources;
     }
 
     public static ResourceSpans getResourceSpans(final String serviceName, final String spanName, final byte[]
-            spanId, final byte[] parentId, final byte[] traceId, final Span.SpanKind spanKind) {
+            spanId, final byte[] parentId, final byte[] traceId, final Span.SpanKind spanKind, final String endTime,
+                                                 final Long durationInNanos, final Integer statusCode) {
         final ByteString parentSpanId = parentId != null ? ByteString.copyFrom(parentId) : ByteString.EMPTY;
+        final long endTimeInNanos = convertTimeStampToNanos(endTime);
+        final long startTimeInNanos = new BigDecimal(endTimeInNanos).subtract(new BigDecimal(durationInNanos)).longValue();
```

Comment:
Can we not just subtract longs directly without needing a conversion?

Reply:
Good catch. This is an overkill